src/adapters/morpho-aave-v2/common/morphoBasePoolAdapter.ts

```diff
@@ -0,0 +1,633 @@
+import { formatUnits } from 'ethers'
+import { AdaptersController } from '../../../core/adaptersController'
+import { Chain } from '../../../core/constants/chains'
+import { RAY } from '../../../core/constants/RAY'
+import { SECONDS_PER_YEAR } from '../../../core/constants/SECONDS_PER_YEAR'
+import { ZERO_ADDRESS } from '../../../core/constants/ZERO_ADDRESS'
+import { IMetadataBuilder } from '../../../core/decorators/cacheToFile'
+import { NotImplementedError } from '../../../core/errors/errors'
+import { aggregateTrades } from '../../../core/utils/aggregateTrades'
+import { aprToApy } from '../../../core/utils/aprToApy'
+import { CustomJsonRpcProvider } from '../../../core/utils/customJsonRpcProvider'
+import { getTokenMetadata } from '../../../core/utils/getTokenMetadata'
+import { logger } from '../../../core/utils/logger'
+import { formatProtocolTokenArrayToMap } from '../../../core/utils/protocolTokenToMap'
+import {
+  BaseTokenMovement,
+  GetPositionsInput,
+  GetEventsInput,
+  GetApyInput,
+  GetAprInput,
+  GetTotalValueLockedInput,
+  GetProfitsInput,
+  GetConversionRateInput,
+  MovementsByBlock,
+  PositionType,
+  ProtocolAdapterParams,
+  ProtocolDetails,
+  ProtocolTokenApr,
+  ProtocolTokenApy,
+  ProtocolTokenUnderlyingRate,
+  ProfitsWithRange,
+  ProtocolTokenTvl,
+  ProtocolPosition,
+  TokenBalance,
+  TokenType,
+  Underlying,
+  UnderlyingTokenRate,
+} from '../../../types/adapter'
+import { Erc20Metadata } from '../../../types/erc20Metadata'
+import { Protocol } from '../../protocols'
+import {
+  MorphoAaveV2__factory,
+  MorphoAToken__factory,
+  MorphoAaveV2Lens__factory,
+} from '../contracts'
+import { TransferEvent } from '../contracts/MorphoAToken'
+
+type MorphoAaveV2PeerToPoolAdapterMetadata = Record<
+  string,
+  {
+    protocolToken: Erc20Metadata
+    underlyingToken: Erc20Metadata
+  }
+>
+
+const morphoAaveV2ContractAddresses: Partial<
+  Record<Protocol, Partial<Record<Chain, string>>>
+> = {
+  [Protocol.MorphoAaveV2]: {
+    [Chain.Ethereum]: '0x777777c9898d384f785ee44acfe945efdff5f3e0',
+  },
+}
+
+export abstract class MorphoBasePoolAdapter implements IMetadataBuilder {
+  protocolId: Protocol
+  chainId: Chain
+
+  protected provider: CustomJsonRpcProvider
```

Comment:
convention: prepend all protected/private vars with a `_`
```suggestion
  protected _provider: CustomJsonRpcProvider
```